packages/mds-stream/kafka/agency-stream-kafka.ts

```diff
@@ -36,10 +36,46 @@ export const AgencyStreamKafka: AgencyStreamInterface = {
       tripMetadataProducer.initialize()
     ])
   },
-  writeEvent: eventProducer.write,
-  writeTelemetry: telemetryProducer.write,
-  writeDevice: deviceProducer.write,
-  writeTripMetadata: tripMetadataProducer.write,
+  writeEvent: async msg => {
+    try {
```

Comment:
I am wondering if there’s a way to handle this pattern generically within the `write` method.

Reply:
Yeah, most likely, I was a little spooked about doing some weird self mutation within the write method itself though